injector/network_disruption.go

```diff
@@ -891,17 +912,110 @@ func (i *networkDisruptionInjector) handleFiltersForServices(interfaces []string
 	return nil
 }
 
+// handleFiltersForServices creates tc filters on given interfaces for hosts in disruption spec classifying matching packets in the given flowid
+func (i *networkDisruptionInjector) handleFiltersForHosts(interfaces []string, flowid string) error {
+	hosts := hostsWatcher{}
+
+	hostFilterMap, err := i.addFiltersForHosts(interfaces, i.spec.Hosts, flowid)
+	if err != nil {
+		return err
+	}
+
+	hosts.hostFilterMap = hostFilterMap
+
+	go i.watchHostChanges(interfaces, hosts, flowid)
+
+	return nil
+}
+
+// watchHostChanges watches for changes to the resolved IP for hosts
+func (i *networkDisruptionInjector) watchHostChanges(interfaces []string, hosts hostsWatcher, flowid string) {
```

Comment:
Could we maybe only Clean/Reinject periodically?
Potentially with only a hasChange detection mechanism to guard useless reinjection.

IMO it's best to have 1 way to do things, 1 to inject, 1 to clean.

And we trigger those when relevant.

We might do a little more work (checking services DNS twice) but it's less error prone IMO and less tests and less code to maintain.

WDYT?

Reply:
Discussed on slack, but putting here, I'd prefer to keep this approach.
1. I'd rather be consistent with how watchServiceChanges works
2. Adding another async way of triggering clean and reinject is similarly complex to what we're doing here
3. A clean/reinject approach would leave tiny blips of connectivity during disruptions, everytime we had to update a single filter. I'd prefer to try this approach until it actually proves to be buggy